add potential as parent class #1093

markus-rothkoetter opened the pull request:
closes #1093

markus-rothkoetter commented:
@l-emele, @stap-m
I'd be happy, if one of you could review my draft PR as well as commits.
Just to make sure that I didn't forget anything before eventually pull requesting.

I also have to remaining workflow questions:

**Question 1:**
Is it necessary to add the term-tracker information to the moved classes? 
  ('flow potential' and 'stock potential')

The wiki/term-tracker-item states 
> If there are already term tracker annotations for this term, add your term track AT THE TOP of the annotation.
But I'm unsure if this applies to restructurings as well ... 

**Question 2:**
Protégé also updated the timestamp in 'catalog-v001.xml'. 
Shall these changes be staged to? 
If yes, I'll add a commit.
(I read in the documentation that it's used as config for imports, so I was wondering if this is rather a local thing.)

l-emele reviewed (changes requested):
I added @stap-m and myself as reviewers and pressed the "Ready for review" button.

> Question 1:
Is it necessary to add the term-tracker information to the moved classes?
('flow potential' and 'stock potential')

Yes, the term tracker needs to be updated. But more importantly, also the definitions need slight updates. As these are now subclasses of `potential`, the definition has to be now: _A flow potential is a **potential** that ..._ and _A stock potential is a **potential** that ..._ Also additions/deletions/changes have to be documented in CHANGELOG.md (simply naming the class labels and PR number in the respective categories of the "unreleased" section.

Please be aware, that the wiki entry did not reflect, what we really to regarding the term tracker items. I changed it just now. We add new entries **at the bottom**.

> Question 2:
Protégé also updated the timestamp in 'catalog-v001.xml'.
Shall these changes be staged to?
If yes, I'll add a commit.
(I read in the documentation that it's used as config for imports, so I was wondering if this is rather a local thing.)

I am always unsure what to do with the catalog file. @jannahastings : Shall these changes be included or not?

jannahastings commented:
> I am always unsure what to do with the catalog file. @jannahastings : Shall these changes be included or not?

The only time that changes to the catalog need to be committed are if we add new files, e.g. new modules, or change the way that the files are structured in directories. Otherwise, ignore it (perhaps add to your .gitignore?)

markus-rothkoetter commented:
> Yes, the term tracker needs to be updated. 

I updated it accordingly.
Nevertheless, I'm asking myself why the changes shall be appended at the bottom.

Isn't it more reasonable to have the changes at the top in order to see the most recent ones at first?

>But more importantly, also the definitions need slight updates. As these are now subclasses of potential, the definition has to be now: A flow potential is a potential that ... and A stock potential is a potential that ... 

Done. (Hope so ...)

I adjusted it, so that all redundant information covered by the parent class are removed e.g.:

````
A flow potential is a potential that describes the input or output of a process per time unit. 
````

... is the updated version of ...

````
A flow potential is a quantity value that describes 
the upper limit of an input or output of a process in a spatial region of reference per unit time.
````

... based on ...

````
A potential is a quantity value that describes some upper limit in a spatial region of reference.
````

Is this how it is supposed to be?

>Also additions/deletions/changes have to be documented in CHANGELOG.md (simply naming the class labels and PR number in the respective categories of the "unreleased" section.

Ah, this was a obvious one that I overlooked.

markus-rothkoetter commented:
> > I am always unsure what to do with the catalog file. @jannahastings : Shall these changes be included or not?
> 
> The only time that changes to the catalog need to be committed are if we add new files, e.g. new modules, or change the way that the files are structured in directories. Otherwise, ignore it (perhaps add to your .gitignore?)

I also added a "quick tip" to the `CONTRIBUTING.md` stating this.
In case this is no good idea, we can simply drop the specific commit.

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -6938,13 +6953,17 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/591",
 
     Annotations: 
-        <http://purl.obolibrary.org/obo/IAO_0000115> "A flow potential is a quantity value that describes the upper limit of an input or output of a process in a spatial region of reference per unit time.",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "A flow potential is a potential that describes the input or output of a process per time unit.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "A flow potential is a potential of an input or output of a process, per time unit.",
```

stap-m replied:
I'd put it even shorter...

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -7010,13 +7029,17 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/607",
 
     Annotations: 
-        <http://purl.obolibrary.org/obo/IAO_0000115> "A stock potential is a quantity value that describes the upper limit of a stock of a source or sink in a spatial region of reference.",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "A stock potential is a potential that describes the stock of a source or sink.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "A stock potential is a potential of the stock of a source or sink.",
```

stap-m reviewed (changes requested):
The rest looks fine!

markus-rothkoetter commented:
> I'd put it even shorter...

Perfect. I was unsure how much simpler it shall be.

markus-rothkoetter commented:
@l-emele Could you re-review this PR?
(I'm not sure if my re-request worked)

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -6956,13 +6971,17 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/591",
     
     SubClassOf: 
-        OEO_00000350
+        <http://openenergy-platform.org/ontology/oeo/oeo-physical/OEO_00290000>
```
I think, this should be simply `OEO_00290000`.

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -7028,13 +7047,17 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/607",
     
     SubClassOf: 
-        OEO_00000350
+        <http://openenergy-platform.org/ontology/oeo/oeo-physical/OEO_00290000>
```
Ditto.

markus-rothkoetter commented:
@l-emele 
The proposed adjustments unfortunately break the `oeo-physical` import.

I verified the online failure with an offline test in Protégé – same result.
As I did my last adjustment in the GitHub WebEditor, I also crossed checked, if the error persists when using a offline editor – same result.

All my previous changes were made in Protégé, so the full iri was inserted by the programm.
Is there a configuration option, that I'm not aware of, to prevent this?

Can you indicate a solution how to incorporate these changes without breaking the `oeo-physical`-file?

l-emele commented:
Hm, not good if this brakes things. I just looked through the whole file. In most cases, there is just the ID (like `OEO_0000abcd`), but in few cases there are there are the full IRIs (like `<http://openenergy-platform.org/ontology/oeo/oeo-physical/OEO_00010239>`). @jannahastings : Any idea, why this is?

If we don't find a quick solution, I suggest that you ignore my last comment and revert to the working version with the IRIs.

markus-rothkoetter commented:
I think this relates to the "specified IRI" from Protégé.
While working on #1112 I noticed that the active IRI was inserted.
I'll come back to this and test if this is the case after finishing #1112.
